csharp-mode.el

```diff
@@ -285,12 +284,27 @@ casts and declarations are fontified.  Used on level 2 and higher."
           ;; merge with cc-mode defaults
           (c-lang-const c-basic-matchers-after)
 
+          ;; Hack: Last prop-identifier
+          `(("\\.\\([A-Z][A-Za-z0-9_]+\\)" 1 font-lock-variable-name-face t))
+          `(("^using\\s *\\([A-Z][A-Za-z0-9_]+\\);" 1 font-lock-variable-name-face t))
+          `(("^namespace\\s *\\([A-Z][A-Za-z0-9_]+\\)\\s *" 1 font-lock-variable-name-face t))
+
           ;; function names
-          `(("\\.\\([A-Za-z0-9_]+\\)[<(]" 1 font-lock-function-name-face t))
+          `(("\\([A-Za-z0-9_]+\\)\\(<[a-zA-Z0-9, ]+>\\)?(" 1 font-lock-function-name-face t))
+
+          ;; class names with inheritance
+          `(("\\<\\([A-Z][A-Za-z0-9_]+\\)\\s *:" 1 font-lock-type-face t))
+
+          ;; Single identifier in attribute
+          `(("\\[\\([A-Z][A-Za-z0-9_]*\\)\\][^;]" 1 font-lock-variable-name-face t))
+
+          ;;  Types after 'new'
+          `((,(concat "new\\s *\\([" (c-lang-const c-symbol-chars) "]+\\)")
+             1 font-lock-type-face t))
           ))
 
 (defcustom csharp-font-lock-extra-types
-  (list (concat "[" c-upper "]\\sw*[" c-lower "]\\sw"))
+  (list "[A-Z][a-zA-Z0-9_]*")
```

Comment:
Another place where we might benefit from having a regexp constant for type-names.

Reply:
Added in 91c0f3c